### DISCUSSION TITLE
client.get_activities returning None for activity.laps, but there are laps

### DISCUSSION THREAD

**Discussion by blairjbass** [hozn/stravalib#494] [Q&A]
Hi all,

I am new to playing around with the strava api and stravalib.

I have noticed when I call client.get_activities() that I get **None** for the laps field for every single activity. However, if I explicitly call client.get_activity_laps(), I get the laps.

I am quickly reaching my short term limits having to call  client.get_activity_laps() for every single activity. 

I am just wondering if anyone else has run into this issue.  

Here's is sample code to illustrate the issue:

```
from stravalib import Client
import subprocess

class Strava:
    
    def __init__(self, MY_STRAVA_CLIENT_ID='_redacted_', MY_STRAVA_CLIENT_SECRET='_redacted_'):
        self.MY_STRAVA_CLIENT_ID = MY_STRAVA_CLIENT_ID
        self.MY_STRAVA_CLIENT_SECRET = MY_STRAVA_CLIENT_SECRET
        self.access_token = None
        self.client = Client()

    def oath(self):
        url = self.client.authorization_url(client_id=self.MY_STRAVA_CLIENT_ID, redirect_uri='http://127.0.0.1:5000/authorization', scope=['read_all','profile:read_all','activity:read_all'])
        print(url)
        subprocess.run("clip", text=True, input=url)
        CODE = input("The above URL has been copied to clipboard. Go to it, and copy and paste the code: ")
        self.access_token = self.client.exchange_code_for_token(client_id=self.MY_STRAVA_CLIENT_ID, client_secret=self.MY_STRAVA_CLIENT_SECRET, code=CODE)
    
    def test_laps(self):
        activities = self.client.get_activities(before="2024-05-02", after="2024-05-01")
        for activity in activities:
            if activity.laps is None:
                print(f"Activity ID {activity.id}: client.get_activities() returned NULL for activity.laps. ")
            laps = self.client.get_activity_laps(activity.id)
            n_laps = sum([1 for lap in laps])
            print(f"Activity ID {activity.id}: client.get_activity_laps() returns {n_laps} laps.")
            

s = Strava()
s.oath()
s.test_laps()

```
Ouput:

```
No rates present in response headers
Activity ID 11311490103: client.get_activities() returned NULL for activity.laps.
Activity ID 11311490103: client.get_activity_laps() returns 18 laps.
```

**Comment by lwasser** [comment]
hey there @blairjbass 👋 i'm just responding here to let you know that we've seen this post and will try to get back to you soon. work life has been busy!

**Comment by jsamoocha** [comment]
Hi,

when you call client.get_activities(), Strava will respond with a list of SummaryActivity objects (see https://developers.strava.com/docs/reference/#api-Activities-getLoggedInAthleteActivities). These don't have a laps property.

Unfortunately, there's no other way to retrieve laps than separately for each activity (which indeed uses a lot of requests).

We can improve our (stravalib) API by being more specific about the response types (e.g., SummaryActivity vs DetailedActivity). We're currently working on a major upgrade with breaking changes and might add this improvement as well.
